stacker/actions/build.py

```diff
@@ -270,6 +269,9 @@ def pre_run(self, outline=False, dump=False, *args, **kwargs):
                 provider=self.provider,
                 context=self.context)
 
+        # Clear old sns events
+        self.provider.pre_run()
```

Comment:
I wouldn't use `pre_run` for this, instead I'd name it what it is/what it actually does.